src/parse.js

```diff
@@ -87,12 +99,25 @@ const PARSERS = {
 };
 
 /*
- * Returns attributes (with or without k)
+ * Returns attributes (with or without key)
  */
-function getAttributes(model: SlateModel, options: Options): Object {
+function getAttributes(
+    model: SlateModel,
+    options: Options,
+    // True to spread the data as attributes.
+    // False to keep it under `data` and to make `type` explicit
+    asShorthand: boolean = true
+): Object {
     return {
+        // type
+        ...(!asShorthand && model.type ? { type: model.type } : {}),
+        // key
         ...(options.preserveKeys && model.key ? { key: model.key } : {}),
-        ...model.data.toJSON()
+        // data
+        ...(!asShorthand && !model.data.isEmpty()
+            ? { data: model.data.toJSON() }
+            : model.data.toJSON()),
+        ...(!asShorthand && model.isVoid ? { isVoid: true } : {})
```

Comment:
Could you split it into an if and a return value? Right now, I can hardly tell what's happening.